core/services/DisplayScreen.py

```diff
@@ -630,14 +645,15 @@ def _update_predictions(self):
         self.detected_objects = []
         if self.objects_detector_prediction:
             self.detected_objects.extend(self.objects_detector_prediction)
+            self._update_detected_objects(self.detected_objects)
         if self.firearm_detector_prediction:
             self.detected_objects.extend(self.firearm_detector_prediction)
-
-        self._update_detected_objects(self.detected_objects)
+            self._update_detected_objects(self.detected_objects)
```

Comment:
Add separate condition for detected objects. Don't update it within these conditions as it leads to wrong output.